Multisig/App/RemoteNotificationHandler.swift

```diff
@@ -232,24 +197,44 @@ class RemoteNotificationHandler {
             let sig = try key.sign(hash: hash)
             return sig.hexadecimal
         }
-        return (hashPreimage, hash.toHexStringWithPrefix(), timestamp, signatures)
+        return (hashPreimage, hash.toHexStringWithPrefix(), signatures)
     }
 
     private func unregister(address: Address, networkId: String) {
-        queue.async { [unowned self] in
-            do {
-                try SafeTransactionService.unregister(deviceID: self.storedDeviceID!,
-                                                      address: address,
-                                                      networkId: networkId)
-            } catch {
-                logError("Failed to unregister device", error)
-            }
-        }
+        App.shared.clientGatewayService.unregister(deviceID: self.storedDeviceID!,
+                                              address: address,
+                                              networkId: networkId)
     }
 
     private func registerAll() {
-        let addresses = Safe.all.compactMap { $0.address }.compactMap { Address($0) }
-        register(addresses: addresses)
+        guard let token = token else { return }
+        guard let deviceID = storedDeviceID?.lowercased() else {
+            assertionFailure("Programmer error: missing device ID")
+            return
+        }
+
+        do {
+            let appConfig = App.configuration.app
+            let timestamp = String(format: "%.0f", Date().timeIntervalSince1970)
+
+            var safeRegistration: [SafeRegistration] = []
+            for networkSafes in Network.networkSafes() {
+                let safes = networkSafes.safes.compactMap { $0.address }.compactMap { Address($0) }.map { $0.checksummed }.sorted()
+                let signResult = try Self.sign(safes: safes, deviceID: deviceID, token: token, timestamp: timestamp)
```

Comment:
do we need to check for empty `safes` and `continue` in that case?

Reply:
I have made same check in different way, please check